### PR TITLE
feat(clerk-js): Add allowedRedirectProtocols

### PR DESCRIPTION
**Pull request by BRKalow**
## Description

Introduce a new Clerk option, `allowedRedirectProtocols` that allows users to specify custom protocols to be allowed as part of user-provided redirect URLs. This is necessary for cases where deep-linking is used, in an Electron app for example.

As part of this, I've removed the protocol check from `windowNavigate`. This is an internal method that is used in only a few places, none of which should be accepting arbitrary, user-provided URLs. I've also added a JSDoc comment indicating `windowNavigate` should never be used when accepting user-provided URLs.

fixes SDKI-782

Closes: https://github.com/clerk/javascript/issues/4667

## Checklist

- [x] `pnpm test` runs as expected.
- [x] `pnpm build` runs as expected.
- [x] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/clerk-js/src/core/__tests__/clerk.test.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ describe('Clerk singleton', () => {
   const productionPublishableKey = 'pk_live_Y2xlcmsuYWJjZWYuMTIzNDUucHJvZC5sY2xjbGVyay5jb20k';
 
   const mockNavigate = jest.fn((to: string) => Promise.resolve(to));
-  const mockedLoadOptions = { routerPush: mockNavigate, routerReplace: mockNavigate };
+  const mockedLoadOptions = { routerDebug: true, routerPush: mockNavigate, routerReplace: mockNavigate };
```
Comment (alexcarpenter) [inline packages/clerk-js/src/core/__tests__/clerk.test.ts:51]: Is this addition intentional or just for local testing purposes?

**Review by alexcarpenter — APPROVED** [review]

File: packages/clerk-js/src/core/__tests__/clerk.test.ts (hunk 1 above)
Comment (BRKalow) [reply packages/clerk-js/src/core/__tests__/clerk.test.ts:51]: This is just for the unit tests 👍 The option already exists

File: packages/clerk-js/src/utils/windowNavigate.ts [hunk 2]
```diff
@@ -8,16 +11,13 @@ export const ALLOWED_PROTOCOLS = [
   'chrome-extension:',
 ];
 
+/**
+ * Helper utility to navigate via window.location.href. Also dispatches a clerk:beforeunload custom event.
+ *
+ * Note that this utility should **never** be called with a user-provided URL. We make no specific checks against the contents of the URL here and assume it is safe. Use `Clerk.navigate()` instead for user-provided URLs.
+ */
```
Comment (panteliselef) [inline packages/clerk-js/src/utils/windowNavigate.ts:18]: Should we replace the usage for `windowNavigate` from SignIn.ts and SignUp.ts with `Clerk.navigate()` ?

**Review by panteliselef — APPROVED** [review]

File: packages/clerk-js/src/utils/windowNavigate.ts (hunk 2 above)
Comment (BRKalow) [reply packages/clerk-js/src/utils/windowNavigate.ts:18]: I don't think we need to, as the url is provided from our API and it will always be external.
